src/sys.rs

```diff
@@ -0,0 +1,425 @@
+
+pub use self::inner::*;
+
+#[cfg(unix)]
+mod inner {
+    use libc::{self, c_int, c_long, c_char, time_t};
+    use ::Tm;
+
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
+    pub use self::mac::*;
+    #[cfg(all(not(target_os = "macos"), not(target_os = "ios")))]
+    pub use self::unix::*;
+
+    /// ctime's `tm`
+    #[repr(C)]
+    struct tm {
+        tm_sec: c_int,
+        tm_min: c_int,
+        tm_hour: c_int,
+        tm_mday: c_int,
+        tm_mon: c_int,
+        tm_year: c_int,
+        tm_wday: c_int,
+        tm_yday: c_int,
+        tm_isdst: c_int,
+        tm_gmtoff: c_long,
+        tm_zone: *const c_char,
+    }
+
+    impl Default for tm {
+        fn default() -> tm {
+            tm {
+                tm_sec: 0,
+                tm_min: 0,
+                tm_hour: 0,
+                tm_mday: 0,
+                tm_mon: 0,
+                tm_year: 0,
+                tm_wday: 0,
+                tm_yday: 0,
+                tm_isdst: 0,
+                tm_gmtoff: 0,
+                tm_zone: 0 as *const c_char
+            }
+        }
+    }
+
+    impl<'a> From<&'a Tm> for tm {
+        fn from(tm: &'a Tm) -> tm {
+            tm {
+                tm_sec: tm.tm_sec,
+                tm_min: tm.tm_min,
+                tm_hour: tm.tm_hour,
+                tm_mday: tm.tm_mday,
+                tm_mon: tm.tm_mon,
+                tm_year: tm.tm_year,
+                tm_wday: tm.tm_wday,
+                tm_yday: tm.tm_yday,
+                tm_isdst: tm.tm_isdst,
+                tm_gmtoff: tm.tm_utcoff as c_long,
+                .. tm::default()
+            }
+        }
+    }
+
+    fn tm_to_rust_tm(tm: &tm, rust_tm: &mut Tm) {
```

Comment:
I've added the `utcoff` to this function. I left `nsec` in lib.rs, since it's not useful to any of the ffi functions, and instead of having to set `tm.tm_nsec` in 4 different functions (unix and windows), it can just be in 2 functions (`at` and `at_utc`).